### PR TITLE
Feature/jenkins integration

### PR DESCRIPTION
**Pull request by askldjd**
Jenkinsfile is executed when Jenkins detects changes on specific branch. Currently, Jenkins is monitoring features/jenkins-integration, and I would like to integrate with master.

The script performs the following:
 - Notify Slack on build start (see channel #appeals-deployment)
 - Checkout appeals-deployment for ansible scripts
 - Run the deploy AMI playbook
 - Notify Slack for build completion.

The script has been tested in the branch, but has not been tested with master.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+      // Token to access the appeals deployment repo.
+      $class: 'StringBinding',
+      credentialsId : 'GIT_CREDENTIAL',
+      variable: 'GIT_CREDENTIAL',
+    ],
+    [
+      // Used to decrypt the Ansible vault pass files.
+      $class: 'StringBinding',
+      credentialsId : 'VAULT_PASS',
+      variable: 'VAULT_PASS',
```
Comment (shanear) [inline Jenkinsfile]: ooh, the fact that we store this on the jenkins server means we should have it mega locked down.

Comment (askldjd) [inline Jenkinsfile]: What is considered mega locked down?

Right now, Jenkins master has no public accessible port. Within Jenkins, the VAULT_PASS is stored in the Jenkins credential plugin as a secret text. The [credential plugin](https://wiki.jenkins-ci.org/display/JENKINS/Credentials+Plugin) prevents the secret text from being shown anywhere in logs. Is that good enough?

Comment (shanear) [inline Jenkinsfile]: @askldjd Awesome. That seems good enough for me.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+          sh 'git submodule init'
+          sh 'git submodule update'
+        }
+      }
+
+      // Execute the ansible playbook to deploy the AMI. This stage will take
+      // awhile to complete.
+      stage ('deploy') {
+        dir ('./appeals-deployment') {
+          sh "nohup -- echo \"${env.VAULT_PASS}\" </dev/null 2>&1 | \
```
Comment (orischwartz) [inline Jenkinsfile]: Does Jenkins automatically log output of builds? If so this command can be simplified to just:

```
         sh "ansible-playbook deploy-to-aws.yml \
              --verbose \
              -i localhost \
              -e app_name=${APP_NAME} \
              -e deploy_env=${APP_ENV} \
              -e app_version=${APP_VERSION} \
              --vault-password-file=${env.VAULT_PASS}"
```

**Comment by askldjd** [comment]
The playbook command is simplified to 
```sh
          sh "echo \"${env.VAULT_PASS}\"| \
            ansible-playbook deploy-to-aws.yml \
              --verbose \
              -i localhost \
              -e app_name=${APP_NAME} \
              -e deploy_env=${APP_ENV} \
              -e app_version=${APP_VERSION} \
              --vault-password-file=/bin/cat"
```

**Pull request merged**